### DISCUSSION TITLE
Typescript schema based client: pocketbase-ts

### DISCUSSION THREAD

**Discussion by Solaris9** [pocketbase/pocketbase#1155] [Show and tell]
Hi everyone, I have been making a schema based client that is strongly typed with Typescript. Currently it's not done but I believe it's far enough to be tested, improved, and worked on until it's finished. Basic CRUD operations work and authentication methods should work. As the readme mentions there are some things missing such as Realtime, Settings, and Logs. Please let me know what you think of it and how I can improve it, or if anyone would like to help contribute to it then I'd appreciate it. Thank you.

GitHub: https://github.com/Solaris9/pocketbase-ts

**Comment by dschissler** [comment]
Perhaps make the `init` function so that it can accept an existing client object in addition to a string.

**Reply by Solaris9** [reply]
Do you mean an existing client from the pocketbase/js-sdk client? I had thought about just wrapping it but I would have rewritten/wrapped everything.

**Reply by dschissler** [reply]
Since you're using a singleton, it would be good to be able to reuse a singleton from elsewhere.  So `init` take either a string or an existing client object.

```typescript
init("https://pocketbase.io")
```

**Comment by benallfree** [comment]
Just added this to [awesome-pocketbase](https://github.com/benallfree/awesome-pocketbase/blob/main/README.md) 🎡
